vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -609,6 +609,11 @@ func resourceVSphereVirtualMachineUpdate(d *schema.ResourceData, meta interface{
 		return err
 	}
 	// Only carry out the reconfigure if we actually have a change to process.
+	cv := virtualmachine.GetHardwareVersionNumber(vprops.Config.Version)
+	tv := d.Get("hardware_version").(int)
```

Comment:
What's the default value if there is no hardware version set? 0 ?

Reply:
Correct. The default value is 0.